src/__tests__/zeebe/integration/Worker-CompleteWithResult.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { restoreZeebeLogging, suppressZeebeLogging } from '../../../lib'
+import { ZeebeGrpcClient } from '../../../zeebe'
+import { cancelProcesses } from '../../../zeebe/lib/cancelProcesses'
+import { CreateProcessInstanceResponse } from '../../../zeebe/lib/interfaces-grpc-1.0'
+
+jest.setTimeout(30000)
+suppressZeebeLogging()
+
+const zbc = new ZeebeGrpcClient()
+let wf: CreateProcessInstanceResponse | undefined
+let processDefinitionKey1: string
+let bpmnProcessId1: string
+
+beforeAll(async () => {
+	const res1 = await zbc.deployResource({
+		processFilename: './src/__tests__/testdata/Worker-JobResult.bpmn',
+	})
+	;({
+		processDefinitionKey: processDefinitionKey1,
+		bpmnProcessId: bpmnProcessId1,
+	} = res1.deployments[0].process)
+	await cancelProcesses(processDefinitionKey1)
+})
+
+afterEach(async () => {
+	if (wf?.processInstanceKey) {
+		await zbc.cancelProcessInstance(wf.processInstanceKey).catch((e) => e)
+	}
+})
+
+afterAll(async () => {
+	await zbc.close()
+	await cancelProcesses(processDefinitionKey1)
+	restoreZeebeLogging()
+})
+
+test('Can complete a task with job corrections', (done) => {
+	zbc
+		.createProcessInstance({
+			bpmnProcessId: bpmnProcessId1,
+			variables: {},
+		})
+		.then((res) => {
+			wf = res
+			zbc.createWorker({
+				taskType: 'job-correction',
+				taskHandler: async (job) => {
+					expect(job.processInstanceKey).toBe(wf?.processInstanceKey)
+					const res1 = await job.completeWithJobResult({
+						variables: {},
+						result: {
+							denied: true,
+						},
```

Comment:
💭 Given the test case (the process has a service task for which this job worker code will run) this code may eventually break.

Job result is at this time only meant to be used for [User Task Listeners](https://docs.camunda.io/docs/next/components/concepts/user-task-listeners/).

For now, we allow regular job completions with a job result ignoring the result. However, in the future we may add validations against it.

🔧 As a more durable test case, please consider a `Camunda User Task` with a `completing` task listener. You can then try to complete the user task, which creates a job that you can activate and complete with a job result.